### PR TITLE
Add UNet model config and refactor diffusion model.

### PR DESCRIPTION
**Pull request by yichunk**
Add UNet model config and refactor diffusion model.
`convert_to_tflite.py` of stable diffusion works, and the pipeline generates reasonable result.

BUG=b/311216181

### CODE REVIEW

File: ai_edge_torch/generative/layers/attention.py [hunk 1]
```diff
@@ -299,30 +301,42 @@ def forward(
...
       input_pos: Optional[torch.Tensor] = None,
   ):
-    B, T, E = x.size()  # batch size, sequence length, embedding dimensionality (n_embd)
+    """Forward function of the CrossAttention layer.
+
+    Args:
+      x (torch.Tensor): the target tensor, with shape [B, target_seq_len, ...]
+      y (torch.Tensor): the source tensor, with shape [B, source_seq_len, ...]
+      rope (Tuple[torch.Tensor, torch.Tensor]): the input rope tensor.
+      input_pos (torch.Tensor): the optional input position tensor.
```
Comment (haozha111) [inline ai_edge_torch/generative/layers/attention.py:310]: for rope, the comment needs to mention it's optional. and also add `mask` here?

File: ai_edge_torch/generative/layers/attention.py [hunk 2]
```diff
@@ -299,30 +301,42 @@ def forward(
...
+
+    Returns:
+      output activation from this self attention layer.
```
Comment (haozha111) [inline ai_edge_torch/generative/layers/attention.py:313]: from this cross attention layer.

File: ai_edge_torch/generative/layers/unet/blocks_2d.py [hunk 3]
```diff
@@ -147,7 +154,11 @@ def __init__(self, config: unet_cfg.CrossAttentionBlock2DConfig):
     self.config = config
     self.norm = layers_builder.build_norm(config.query_dim, config.normalization_config)
     self.attention = CrossAttention(
-        config.query_dim, config.cross_dim, config.attention_config, 0, True
+        config.query_dim,
+        config.cross_dim,
+        config.attention_config,
+        0,
+        enable_hlfb=True,
```
Comment (haozha111) [inline ai_edge_torch/generative/layers/unet/blocks_2d.py:161]: can you read this `enable_hlfb` from Model config class instead of setting to True always?

Comment (yichunk) [reply ai_edge_torch/generative/layers/unet/blocks_2d.py]: done

File: ai_edge_torch/generative/layers/attention.py (hunk 2 above)
Comment (yichunk) [reply ai_edge_torch/generative/layers/attention.py]: done

File: ai_edge_torch/generative/layers/attention.py (hunk 1 above)
Comment (yichunk) [reply ai_edge_torch/generative/layers/attention.py]: done

**Review by haozha111 — DISMISSED** [review]

**Review by haozha111 — APPROVED** [review]

**Pull request merged**